godot-core/src/registry/class.rs

```diff
@@ -228,14 +238,22 @@ pub fn auto_register_classes(init_level: InitLevel) {
             editor_plugins.push(info.class_name);
         }
 
+        if let Some(register_singleton_fn) = info.register_singleton_fn {
+            singletons.push(register_singleton_fn)
+        }
+
         register_class_raw(info);
 
         out!("Class {class_name} loaded.");
     }
 
-    // Will imminently add given class to the editor.
+    for register_singleton_fn in singletons {
+        register_singleton_fn()
+    }
+
+    // Will imminently add given class to the editor in Godot before 4.4.1.
     // It is expected and beneficial behaviour while we load library for the first time
-    // but (for now) might lead to some issues during hot reload.
+    // but might lead to some issues during hot reload.
```

Comment:
What do you mean with 4.4.1 -- what happens after that version?

Further, I know the comment about hot-reload issues was there before, but do you think we can concretize it?